### PR TITLE
feat(rubygems): Support `info` endpoint

### LINKED ISSUE #15459: Support Nexus repository manager for RubyGems
### What would you like Renovate to be able to do?

Nexus repository manager doesn't support the '/api/v1/' pattern for RubyGem discovery, however, `bundler` does support installing Gems when the Nexus URL is used as the `source` in the `Gemfile`.

When Renovate finds a Gemfile with a source of a source of Nexus the logs show:

```shell
DEBUG: Datasource connection error (repository=example)
       "datasource": "rubygems",
       "packageName": "example",
       "url": "https://private/repository/example/api/v1/gems/example.json",
       "errCode": "ERR_NON_2XX_3XX_RESPONSE"
DEBUG: Failed to look up dependency ***
```

The above URL would result in a 400.

### If you have any ideas on how this should be implemented, please tell us here.

The `source` used for `bundler` is just a URL link to the Gem group rather than an API endpoint. Should maybe take that into account.

### Is this a feature you are interested in implementing yourself?

No

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

- It makes sense to query `/versions` endpoint and incrementally update its data, when registry is `rubygems.org`
  - Approach before this PR just skips its caching, but it may cause long processing time
- Otherwise, it's better to query `/info/:package` endpoint for versions list
- When these are failed, then we try another endpoints

## Context

- Closes: #15459

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/rubygems/metadata-cache.ts [hunk 1]
```diff
@@ -63,7 +62,6 @@ export class MetadataCache {
       );
       return data;
     } catch (err) {
-      logger.debug({ err }, 'Rubygems: failed to fetch metadata');
```
Comment (zharinov) [inline lib/modules/datasource/rubygems/metadata-cache.ts:66]: I think people will take this log message too seriously and report it as a bug.
We can do nothing about servers that don't support endpoints with metadata.

**Comment by zharinov** [comment]
PR obtained from local Nexus mirror: https://github.com/renovate-testing/test-15459-nexus-rubygems/pull/4

**Review by viceice — COMMENTED** [review]
shouldn't we first try the current API and only fallback to his one? we currently only know nexus to not support the newer API.
and we can also detect nexus on http headers to directly fallback to info endpoint.

**Comment by zharinov** [comment]
Making processing logic dependent on proprietary implementation should be the last resort, let's try to reorder call sequence first.

File: lib/modules/datasource/rubygems/metadata-cache.ts (hunk 1 above)
Comment (HonkingGoose) [reply lib/modules/datasource/rubygems/metadata-cache.ts:66]: You could keep the debug message, and improve it to explain why it's not a big problem. You could also lower the log level from `err` to something appropiate.

**Review by rarkins — APPROVED** [review]

File: lib/modules/datasource/rubygems/metadata-cache.ts (hunk 1 above)
Comment (rarkins) [inline lib/modules/datasource/rubygems/metadata-cache.ts:66]: How about:

```
  logger.debug(`No metadata endpoint available for ${registryUrl}`);
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Pull request closed without merging**